src/class-wpml-beaver-builder-data-settings.php

```diff
@@ -56,4 +56,31 @@ public function get_fields_to_save() {
 	}
 
 	public function add_hooks(){}
+
+	/**
+	 * Adds slashes to data going into the database as WordPress
+	 * removes them when we save using update_metadata. This is done
+	 * to ensure slashes in user input aren't removed.
+	 *
+	 * Inspired by `\FLBuilderModel::slash_settings`
+	 *
+	 * @param mixed $data The data to slash.
+	 *
+	 * @return mixed The slashed data.
+	 */
+	private function slash( $data ) {
+		if ( is_array( $data ) ) {
+			foreach ( $data as $key => $val ) {
+				$data[ $key ] = $this->slash( $val );
+			}
+		} elseif ( is_object( $data ) ) {
+			foreach ( $data as $key => $val ) {
+				$data->$key = $this->slash( $val );
+			}
+		} elseif ( is_string( $data ) ) {
+			$data = wp_slash( $data );
```

Comment:
@strategio 

I can see your logic tries to address the case when the input is an object, which means you must also handle the case of an array: since an array's item may also an object, you can't call `wp_slash` for both arrays and strings.

That makes perfectly sense.

However, now you call `wp_slash` only when the input is a string.
And, when the input is a string, `wp_slash` calls just `addslashes` (PHP native function). 

So, I'm not sure if it still makes sense to still call `wp_slash` at all now: you could just call `addslashes`.

Though, I'm not sure you can replace `wp_slash` with `addslashes` in your tests: if that's the case, you can ignore this comment.

Reply:
@andreasciamanna, the truth is that I just copied the logic from the code inside Beaver Builder (`\FLBuilderModel::slash_settings`). Just transformed it in a non-static method.